From 9d127abaed19b66c5b75cf353800cff0b81941b2 Mon Sep 17 00:00:00 2001
From: Sumit Bose <sbose@redhat.com>
Date: Tue, 24 Nov 2009 23:11:28 +0100
Subject: [PATCH] Fixes for init script to meet guidelines

---
 config/ctdb.init |   22 ++++++++++++++--------
 1 files changed, 14 insertions(+), 8 deletions(-)

diff --git a/config/ctdb.init b/config/ctdb.init
index 2e25cf1..50f29f1 100755
--- a/config/ctdb.init
+++ b/config/ctdb.init
@@ -170,7 +170,7 @@ start() {
 
     ctdb ping >/dev/null 2>&1 && {
 	echo $"CTDB is already running"
-	return 1
+	return 0
     }
 
     build_ctdb_options
@@ -213,7 +213,7 @@ start() {
     set_ctdb_variables
 
     return $RETVAL
-}	
+}
 
 stop() {
     echo -n $"Shutting down ctdbd service: "
@@ -260,20 +260,26 @@ stop() {
 restart() {
     stop
     start
-}	
+}
 
 status() {
     echo -n $"Checking for ctdbd service: "
     ctdb ping >/dev/null 2>&1 || {
 	RETVAL=$?
-	echo -n "  ctdbd not running. "
 	case $init_style in
 	    suse)
+		echo -n "  ctdbd not running. "
 		set_retval $RETVAL
 		rc_status -v
 		;;
 	    redhat)
-		echo ""
+		if [ -f /var/lock/subsys/ctdb ]; then
+			echo $"ctdb dead but subsys locked"
+			RETVAL=2
+		else
+			echo $"ctdb is stopped"
+			RETVAL=3
+		fi
 		;;
 	esac
 	return $RETVAL
@@ -290,13 +296,13 @@ case "$1" in
     stop)
   	stop
 	;;
-    restart|reload)
+    restart|reload|force-reload)
   	restart
 	;;
     status)
   	status
 	;;
-    condrestart)
+    condrestart|try-restart)
   	ctdb status > /dev/null && restart || :
 	;;
     cron)
@@ -304,7 +310,7 @@ case "$1" in
   	ctdb status > /dev/null || restart
 	;;
     *)
-	echo $"Usage: $0 {start|stop|restart|status|cron|condrestart}"
+	echo $"Usage: $0 {start|stop|restart|reload|force-reload|status|cron|condrestart|try-restart}"
 	exit 1
 esac
 
-- 
1.6.5.2

